contracts/Edition.sol

```diff
@@ -90,26 +94,29 @@ contract Edition is
         description = _description;
         animationUrl = _animationUrl;
         imageUrl = _imageUrl;
-        editionSize = _editionSize;
-        royaltyBPS = _royaltyBPS;
 
+        uint56 endOfMintPeriod;
         if (_mintPeriodSeconds > 0) {
             // overflows are not expected to happen for timestamps, and have no security implications
             unchecked {
-                endOfMintPeriod = block.timestamp + _mintPeriodSeconds;
+                endOfMintPeriod = uint56(block.timestamp + _mintPeriodSeconds);
             }
         }
+
+        state = EditionState({
+            editionSize: uint56(_editionSize),
+            endOfMintPeriod: endOfMintPeriod,
+            royaltyBPS: uint16(_royaltyBPS),
```

Comment:
i read that this truncates the leading bits (preserving the least significant bits)? is it possible to enforce that the caller pass in uint56? i'm wondering if it would be surprising to pass in (for example) `10000000` and the edition storing something `00`. i'm mostly concerned for people who may stumble upon this and deploy it themselves since it's open source, i think we can be safe about it from the backend :smiley:

Reply:
yes that's a good point, we could revert if the value is too big to store in a uint56

Reply:
in [25822d4](https://github.com/showtime-xyz/nft-editions/pull/24/commits/25822d47bdb2ed9bf8c1766f6240313eaa1288ab)